applications/serial_lte_modem/src/slm_ppp.c

```diff
@@ -263,23 +264,44 @@ static unsigned int ppp_retrieve_mtu(void)
 			 * Because, it must be at least 1280 for IPv6,
 			 * while MTU of IPv4 may be less.
 			 */
-			return MIN(populated_info.ipv6_mtu, sizeof(ppp_data_buf));
-		}
-		if (populated_info.ipv4_mtu) {
+			mtu = MIN(populated_info.ipv6_mtu, sizeof(ppp_data_buf));
+		} else if (populated_info.ipv4_mtu) {
 			/* Set the PPP MTU to that of the LTE link. */
-			return MIN(populated_info.ipv4_mtu, sizeof(ppp_data_buf));
+			mtu = MIN(populated_info.ipv4_mtu, sizeof(ppp_data_buf));
 		}
-	}
-
-	LOG_DBG("Could not retrieve MTU, using fallback value.");
-	BUILD_ASSERT(sizeof(ppp_data_buf) >= CONFIG_SLM_PPP_FALLBACK_MTU);
-	return CONFIG_SLM_PPP_FALLBACK_MTU;
-}
-
-static void ppp_set_mtu(void)
-{
-	const unsigned int mtu = ppp_retrieve_mtu();
 
+		/* Try to populate DNS addresses from PDN */
+		if (populated_info.dns_addr4_primary.s_addr != INADDR_ANY) {
+			/* Populate both My address and peer options
+			 * as these are wrong way in Zephyr (it offers my_option DNS)
+			 */
+			ctx->ipcp.peer_options.dns1_address = populated_info.dns_addr4_primary;
+			ctx->ipcp.peer_options.dns2_address = populated_info.dns_addr4_secondary;
+			ctx->ipcp.my_options.dns1_address = populated_info.dns_addr4_primary;
+			ctx->ipcp.my_options.dns2_address = populated_info.dns_addr4_secondary;
+#if defined(CONFIG_LTE_LC_DNS_FALLBACK_ADDRESS)
```

Comment:
This looks like a very unlikely configuration, although possible. We only have memfault config with lte_lc enabled but I guess there is an unknown number of customers using lte_lc with SLM.

Reply:
True, but these two are the only statically configured DNS addresses that I'm aware of. It makes sense to try to fall back into one of these, if these are configured.